src/client/store/ui/review/hooks/index.ts

```diff
@@ -6,6 +6,27 @@ import { useAppSelector } from '@client/store'
 export const useReviewStatus = (key: string): ReviewStatus =>
   useAppSelector((state) => state.ui.review.status[key] || ({} as ReviewStatus))
 
+export const useOdpReviewSummary = (odpId: string): ReviewStatus => {
+  const statuses = useAppSelector((state) =>
+    Object.values(
+      Object.fromEntries(Object.entries(state.ui.review.status).filter(([key]) => key.includes(`odp-${odpId}-`)))
+    )
+  )
+
+  return statuses.reduce(
+    (curr, acc) => {
+      return {
+        hasUnreadMessages: curr.hasUnreadMessages || acc.hasUnreadMessages,
+        status: curr.status !== MessageTopicStatus.resolved ? MessageTopicStatus.opened : acc.status,
+      }
+    },
+    {
+      hasUnreadMessages: false,
+      status: MessageTopicStatus.resolved,
+    }
+  )
+}
```

Comment:
you're iterating twice, which may be superfluous.
you can reduce all the statuses and conditionally update the accumulator based on if current element is odp